packages/next/src/server/web/spec-extension/adapters/headers.ts

```diff
@@ -0,0 +1,225 @@
+import type { IncomingHttpHeaders } from 'http'
+
+function isHeaders(headers: IncomingHttpHeaders | Headers): headers is Headers {
+  // TODO: maybe use instanceof check?
```

Comment:
Why not?

Reply:
I'll experiment! The issue previously with this was that due to bundling for edge, it could create multiple copies of `Headers` which would mean that the `instanceof` check would always fail. Maybe that isn't the case!